src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -42,28 +44,16 @@
 from pants.engine.fs import CreateDigest, Digest, FileContent, MergeDigests, RemovePrefix
 from pants.engine.process import FallibleProcessResult, Process
 from pants.engine.rules import Get, MultiGet, collect_rules, rule, rule_helper
-from pants.engine.target import CoarsenedTargets, FieldSet, Target
+from pants.engine.target import TransitiveTargets, TransitiveTargetsRequest
 from pants.engine.unions import UnionRule
 from pants.util.logging import LogLevel
 from pants.util.ordered_set import FrozenOrderedSet, OrderedSet
 from pants.util.strutil import pluralize, shell_quote
 
 
-@dataclass(frozen=True)
-class MyPyFieldSet(FieldSet):
```

Comment:
Oops, we defined this both in `mypy/rules.py` and `mypy/subsystem.py`

Reply:
I've been naming FieldSets to be more use-case-specific.

E.g. `MyPyCheckFieldSet`